client/volume_inspect.go

```diff
@@ -22,3 +24,24 @@ func (cli *Client) VolumeInspect(ctx context.Context, volumeID string) (types.Vo
 	ensureReaderClosed(resp)
 	return volume, err
 }
+
+// VolumeInspectWithRaw returns the information about a specific volume in the docker host and it's raw representation
+func (cli *Client) VolumeInspectWithRaw(ctx context.Context, volumeID string) (types.Volume, []byte, error) {
```

Comment:
To remove the code duplicate with `VolumeInspect()` maybe `VolumeInspect()` should call this new method, and just discard the body?

This method does a little bit of extra work, but it shouldn't be significant, and the old method is really just there for backwards compatibility now (it won't actually get called by the engine I believe).

Same comment about the network inspect call.